include/HubFramework/HUBConfig.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2017 Spotify AB.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <Foundation/Foundation.h>
+
+#import "HUBComponentCategories.h"
+
+@protocol HUBActionRegistry;
+@protocol HUBJSONSchema;
+@protocol HUBComponent;
+@protocol HUBComponentFallbackHandler;
+@protocol HUBContentReloadPolicy;
+@protocol HUBComponentLayoutManager;
+@protocol HUBImageLoaderFactory;
+@protocol HUBConnectivityStateResolver;
+@protocol HUBIconImageResolver;
+@protocol HUBViewControllerScrollHandler;
+@protocol HUBComponentRegistry;
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ *  Configuration used to create view controllers when using Hub Framework without `HUBManager`.
+ *
+ *  This allows for using Hub Framework without the `HubManager` and feature registration.
```

Comment:
Could we add a sentence pointing the user to `HUBConfigBuilder`? The first questions that popped into my heads was “how am I supposed to create one of these objects? all the properties are `readonly`?”.